packages/eslint-plugin/src/rules/test-id-consistent-naming.js

```diff
@@ -0,0 +1,67 @@
+"use strict";
+
+const enforceKey = (context, validName) => jsxAttribute => {
+  context.report({
+    node: jsxAttribute,
+    message: `Test ID attribute should be named ${validName}`,
+    fix(fixer) {
+      return fixer.replaceTextRange(jsxAttribute.name.range, validName);
+    },
+  });
+};
+
+module.exports = {
+  create(context) {
+    const pragma = context.options[0] || "data-test-id";
+
+    const match = {
+      'JSXAttribute[name.name = "testId"]': enforceKey(context, pragma),
+      'JSXAttribute[name.name = "data-id-test"]': enforceKey(context, pragma),
+      'JSXAttribute[name.name = "data-idtest"]': enforceKey(context, pragma),
+      'JSXAttribute[name.name = "dataTestId"]': enforceKey(context, pragma),
+      'JSXAttribute[name.name = "datatestid"]': enforceKey(context, pragma),
+    };
+
+    const pragmaMatch =
+      pragma === "data-test-id"
+        ? {
+            'JSXAttribute[name.name = "data-testid"]': enforceKey(
+              context,
+              pragma,
+            ),
+            'JSXAttribute[name.name = "data-testId"]': enforceKey(
+              context,
+              pragma,
+            ),
+            'JSXAttribute[name.name = "data-testID"]': enforceKey(
+              context,
+              pragma,
+            ),
+          }
+        : {
+            'JSXAttribute[name.name = "data-test-id"]': enforceKey(
+              context,
+              pragma,
+            ),
+          };
+
+    return {
```

Comment:
Would it be easier to define a list and filter the chosen attribute name from it, and then generate this object? i.e.

```typescript
const KNOWN_ATTRIBUTE_NAMES = ["testId", "data-test-id", ...];
return KNOWN_ATTRIBUTE_NAMES.filter(name => name !== pragma)
  .reduce((acc, name) => ({
    ...acc,
    [`JSXAttribute[name.name = "${name}"]`]: enforceKey(context, pragma),
  }), {});
```

Reply:
yep sure would